### PR TITLE
Add on-comment trigger to backport action workflow

### PR DESCRIPTION
**Pull request by korthout**
## Description

Adds a workflow configuration to trigger the backport action
when a user writes a `/backport` comment on a pull request.
The bot itself restricts that only merged pull requests can be
backported.

This allows to backport pull requests that have already been merged.
Just add the backport labels to the merged PR and write `/backport`.

> WARNING! The bot will try to backport to the branches of all backport labels. 
> Previously created backport pull requests are not taken into account. 
> New pull requests will be created for each label.

## Related issues

relates to https://github.com/zeebe-io/backport-action/issues/11

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by korthout** [comment]
Since @npepinpe is not in today, @deepthidevaki please have a look

File: .github/workflows/backport-on-comment.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Backport labeled PRs triggered by comment
+on:
+  issue_comment:
+    types: [ created, edited ]
```
Comment (deepthidevaki) [inline .github/workflows/backport-on-comment.yml:4]: Do you want to trigger this also for "edited"?

**Review by deepthidevaki — APPROVED** [review]
Looks good to me. But is there any way to merge both workflows? - most of the code is duplicated, right?

**Comment by korthout** [comment]
Yes, they can be merged. I avoided it because the conditional expression becomes very long, but we also did this for Testbench in https://github.com/zeebe-io/zeebe-cluster-testbench/pull/339. I'll do the same here.

File: .github/workflows/backport-on-comment.yml (hunk 1 above)
Comment (korthout) [reply .github/workflows/backport-on-comment.yml:4]: Probably not a good idea. Thanks for pointing it out

**Review by deepthidevaki — APPROVED** [review]
:tada:

**Comment by korthout** [comment]
bors merge

**Pull request merged**